static/app/components/events/eventTags/eventTagsTree.tsx

```diff
@@ -43,6 +44,10 @@ function addToTagTree(
   originalTag: EventTag
 ): TagTree {
   const BRANCH_MATCHES_REGEX = /\./g;
+  if (!defined(tag.key)) {
+    return tree;
+  }
```

Comment:
shouldn't the backend api which is producing garbage be fixed instead?  this just hides the bug

Reply:
There might be a valid reason for these `null` keys even if I wasn't able to replicate it or test with it myself, it would be hasty to make changes like that since this UI is behind a flag and the previous UI had no issues -- It has been receiving `null` keys for some time with no issues, it'd seem.

This error was triggered from an EA group accessing a new interface I built for the frontend, so this fix provides the best compatibility with users who don't have access to the flags, without changing their experience of the legacy UI.

I will look into where the null keys come from to make sure we can account for it in the new UI, but for now I find this to be the best approach.

Reply:
have you looked into what's causing the `null`?  I really think we should be striving for quality here instead of hacking around our own bugs in the frontend